### PR TITLE
Remove Enrich and Filter for ASP.NET Core metrics

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Towards #https://github.com/open-telemetry/opentelemetry-dotnet/issues/4932
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -86,6 +86,15 @@ metric. This change only affects users setting `OTEL_SEMCONV_STABILITY_OPT_IN`
 to `http` or `http/dup`.
 ([#4934](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934))
 
+* **Breaking**: Removed `Enrich` and `Filter` support for **metrics**
+  instrumentation. The `Enrich` API for `http.server.request.duration` will be
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:90]:
```suggestion
  instrumentation. An `Enrich` API for `http.server.request.duration` will be
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -86,6 +86,15 @@ metric. This change only affects users setting `OTEL_SEMCONV_STABILITY_OPT_IN`
...
+  available for users targeting `.NET8.0` or newer frameworks. For details see
+  [Enrich the ASP.NET Core request
+  metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric).
+  `Filter` capability will not be available for any of the targeted .NET
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:94]:
```suggestion
  `Filter` capability is not available for any .NET
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -86,6 +86,15 @@ metric. This change only affects users setting `OTEL_SEMCONV_STABILITY_OPT_IN`
...
+  versions.
+  ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))
+
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:97]: consider opening a dedicated issue for users to present the case of Filter in metrics. Include a link to that from here, so folks affecting with this change can request for bringing it back/explain the use case etc.

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:97]: https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/CHANGELOG.md#changed You may borrow some ideas from this changelog for breaking change/feedback-issue, if you like that wording!

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:100]: Thanks!. updated.

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 4]
```diff
@@ -86,6 +86,18 @@ metric. This change only affects users setting `OTEL_SEMCONV_STABILITY_OPT_IN`
 to `http` or `http/dup`.
 ([#4934](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934))
 
+* **Breaking**: Removed `Enrich` and `Filter` support for **metrics**
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:89]: I think its best to explicitly mention `AspNetCoreMetricsInstrumentationOptions` is no longer available as the only reason for its existence (enrich/filter) were removed.

**Review by cijothomas — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
referencing the original PR where these options were added https://github.com/open-telemetry/opentelemetry-dotnet/pull/3948

@Temppus FYI

File: src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreMetricsInstrumentationOptions.cs [hunk 5]
```diff
@@ -24,7 +23,7 @@ namespace OpenTelemetry.Instrumentation.AspNetCore;
 /// <summary>
 /// Options for metrics requests instrumentation.
 /// </summary>
-public class AspNetCoreMetricsInstrumentationOptions
+internal class AspNetCoreMetricsInstrumentationOptions
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreMetricsInstrumentationOptions.cs:26]:
```suggestion
internal sealed class AspNetCoreMetricsInstrumentationOptions
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 6]
```diff
@@ -86,6 +86,19 @@ metric. This change only affects users setting `OTEL_SEMCONV_STABILITY_OPT_IN`
...
+  is no longer available. An `Enrich` API for `http.server.request.duration`
+  will be available for users targeting `.NET8.0` or newer frameworks. For
+  details see [Enrich the ASP.NET Core request
+  metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric)
+
+  `Filter` capability is not available for any .NET versions. Please [share your
+  feedback
+  here](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4982) if
+  you are affected.
+  ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:100]:
```suggestion
* **Breaking**: Removed `Enrich` and `Filter` support for **metrics**
  instrumentation. With this change, `AspNetCoreMetricsInstrumentationOptions`
  is no longer available.
  ([#4981](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4981))

  * `Enrich` migration:

    An enrichment API for the `http.server.request.duration` metric is available
    inside AspNetCore for users targeting .NET 8.0 (or newer). For details see:
    [Enrich the ASP.NET Core request
    metric](https://learn.microsoft.com/aspnet/core/log-mon/metrics/metrics?view=aspnetcore-8.0#enrich-the-aspnet-core-request-metric).
  
  * `Filter` migration:

    There is no comparable filter mechanism currently available for any .NET
    version. Please [share your
    feedback](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4982)
    if you are impacted by this feature gap.

    > **Note**
    > The [View
    API](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics/customizing-the-sdk#select-specific-tags)
    may be used to drop dimensions.
```

**Pull request merged**